app/models/developer.rb

```diff
@@ -74,6 +74,16 @@ class Developer < ApplicationRecord
   scope :newest_first, -> { order(created_at: :desc) }
   scope :profile_reminder_notifications, -> { where(profile_reminder_notifications: true) }
   scope :visible, -> { where.not(search_status: :invisible).or(where(search_status: nil)) }
+  scope :with_specialties, ->(specialty_ids) {
+    where_sql = <<-SQL
+      exists (
+        select 1 from specialty_tags
+        where specialty_tags.specialty_id in (?)
+          and specialty_tags.developer_id = developers.id
+      )
+    SQL
```

Comment:
By using `where exists`, a developer record is returned if they have any of the specialties. This works because not only are we getting the specialty_tags that are in the specialty_id array, we are also restricting the specialty_tags to the developers in the outer query.